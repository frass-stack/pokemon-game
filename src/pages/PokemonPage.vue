<template>
    <div>
        <h2>Â¿Quien es ese Pokemon?</h2>
        <PokemonPicture :pokemonId="4" :showPokemon="true" />
        <PokemonOptions :pokemons="pokemonArr" />
    </div>
</template>

<script>
import PokemonOptions from '../components/PokemonOptions.vue'
import PokemonPicture from '../components/PokemonPicture.vue'
import getPokemonsOptions from '@/helpers/getPokemonsOptions'

//console.log(getPokemonsOptions())

export default {
    components: { 
        PokemonPicture,
        PokemonOptions    
    },
    data(){
        return {
            pokemonArr: []
        }
    },
    methods: {
        async mixPokemonArr(){
            this.pokemonArr = await getPokemonsOptions()
        }
    },
    mounted(){
        this.mixPokemonArr()
    }
}
</script>

<style>

</style>