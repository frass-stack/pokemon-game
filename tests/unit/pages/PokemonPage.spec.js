import { shallowMount } from "@vue/test-utils";
import PokemonPage from '@/pages/PokemonPage';
import { pokemons } from "../mocks/pokemon.mocks";

describe('PokemonPage Component', () => {
    let wrapper;
    let mixPokemonArrSpy;
    beforeEach( () => {
        mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArr')
        wrapper = shallowMount( PokemonPage )
    } )

    test('Se debe llamar al metodo "mixPokemonArr" en el mounted', () => {
        expect(mixPokemonArrSpy).toHaveBeenCalled()
    })

    test('Debe hacer snapshot con una vez montado el componente', () => {
        wrapper = shallowMount( PokemonPage, {
            data(){
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Deben existir los componentes', () => {
        wrapper = shallowMount( PokemonPage, {
            data(){
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        const pokemonPicture = wrapper.find('pokemon-picture-stub')
        const pokemonOptions = wrapper.find('pokemon-options-stub')

        expect(pokemonPicture.exists()).toBeTruthy()
        expect(pokemonOptions.exists()).toBeTruthy()

        expect(pokemonOptions.attributes('pokemons')).toBeTruthy()
        expect(pokemonPicture.attributes('pokemonid')).toBe("5")
    })
})