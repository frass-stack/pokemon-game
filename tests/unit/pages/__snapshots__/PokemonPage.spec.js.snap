// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe hacer snapshot con una vez montado el componente 1`] = `
<div>
  <h2>Â¿Quien es ese Pokemon?</h2>
  <pokemon-picture-stub pokemonid="5" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
